.pre-commit-config.yaml

```diff
@@ -0,0 +1,18 @@
+---
+fail_fast: true
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: "v4.4.0"
+    hooks:
+      - id: check-yaml
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: "v1.0.1"
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - numpy
+          - packaging
+          - pandas-stubs
```

Comment:
Not obvious to me why pandas is a dep.

Reply:
https://github.com/cclib/cclib/blob/35d72b0c20696778b5396842e20641541d2e1c7d/cclib/io/ccio.py#L496

Reply:
I should say exactly what this file is. It's technically not necessary for mypy to run.[^1] `.pre-commit-config.yaml` is a configuration file for https://pre-commit.com/, which is two things:
1. A tool that runs linters and fixers from the command line. You do *not* need to have it as a pre-commit hook. Just doing https://pre-commit.com/#4-optional-run-against-all-the-files is enough.
2. A free CI service that runs alongside GH workflows, ReadTheDocs, etc.

The benefit to having 1 is that everyone has a unified way of running all these tools that's independent from CI, specifically opaque GitHub Actions. You can have 1 without 2. 2 would mean that you get the benefits of the hook (clean repository) without forcing contributors to install the hook.

The reason [`pandas-stubs`](https://github.com/pandas-dev/pandas-stubs) appears in `additional_dependencies` is because of how the tool works. For Python-based tools, it creates a fresh venv to run that tool in, and by default it only installs that tool. This means that mypy will not have access to any 3rd-party type information unless the dependencies or packages with only type stubs (for dependencies that don't ship them together) are also installed.

[^1]: For mypy configuration, you only need `pyproject.toml` or `mypy.ini`. I chose the former since everything is moving toward it anyway.